src/Config/ConfigurationLoader.php

```diff
@@ -48,12 +49,34 @@ public function fetchAndStoreConfiguration(?string $flagETag): void
                 syslog(LOG_WARNING, "[Eppo SDK] Empty or invalid response from the configuration server.");
                 return;
             }
-            $fcr = FlagConfigResponse::fromJson($responseData);
+            $fcr = FlagConfigResponse::fromArray($responseData);
             $banditResponse = null;
-            // TODO: Also check current bandit models loaded for optimized bandit loading.
             if (count($fcr->banditReferences) > 0) {
-                $bandits = $this->apiRequestWrapper->getBandits();
-                $banditResponse = new ConfigResponse($bandits->body, date('c'), $bandits->eTag);
+                $canReuseBandits = true;
+                $currentBandits = $currentConfig->getBanditModels();
+
+                foreach ($fcr->banditReferences as $banditKey => $banditReference) {
+                    if (
+                        !array_key_exists(
+                            $banditKey,
+                            $currentBandits
+                        ) || $banditReference->modelVersion !== $currentBandits[$banditKey]
+                    ) {
+                        $canReuseBandits = false;
```

Comment:
Per @leoromanovsky's understandable confusion, let's add some comments here. Perhaps above the `foreach` put something like "We need to fetch bandit models if there are new bandits or any existing bandits have updated model versions" and then maybe above this line something like "There are bandit model versions at play we have not yet loaded"

Reply:
✔️